MatiasDure/Project_IMTS
# Updated the clam interation to using play animation component

**Description**
The clam used to be comprised of two `IInteractable` components, the `Rotate` and `PlayParticle`. Since the designers decided to have a smoother animation than just a basic rotation for the open and close, the clam interaction had to be updated to using the `PlayAnimation` instead of the `Rotate` component. 
Some extra fixes:
* Particles now plays when the clam is completely opened
* Particles don't stop at every interaction, but instead when the clam opens or the duration ends

**Screenshot**
![ClamUpdateToAnimation](https://github.com/user-attachments/assets/fcbe275f-3bb2-47ce-8168-0621357d9eac)

**How To Test**
1. switch the the `update-clam-to-play-animation` branch
2. open the `PlotFrame-Clam` scene
3. Run the project
4. open the plot frame
5. Interact with the clam through the portal
6. Try multiple ways of interaction to check if it always works as expected
